practice-two/src/pages/Analytics/Analytics.jsx

```diff
@@ -65,31 +69,14 @@ const Analytics = () => {
   };
 
   // Fetch data from the server when the component mounts
-  useEffect(() => {
-    const getAllCustomers = async () => {
-      setIsLoading(true);
-      const data = await getAllCustomerService();
-      if (data instanceof Error) {
-        setIsError(data.message);
-        toast.error("Having some error. Please try again!", {
-          position: "top-right",
-          autoClose: 5000,
-          hideProgressBar: false,
-          closeOnClick: true,
-          pauseOnHover: true,
-          draggable: true,
-          progress: undefined,
-          theme: "dark",
-        });
-      } else {
-        setCustomers(data);
-        setIsError(null);
-      }
-      setIsLoading(false);
-    };
-
-    getAllCustomers();
-  }, []);
+  const {
+    data: customers,
+    error: isError,
+    isLoading,
+  } = useSWR(`${BASE_URL}/${PATH}`, getAllCustomerService, {
+    refreshInterval: 5000,
```

Comment:
why need use refreshInterval?

Reply:
Because I want customer data to be updated continuously after 5 second
It just re-render a component list data instead of all page
When I leave page or click to another tag on website, it will stop updating data until I come back to the page
But if I don't leave page, the data will store temporarily  into cache, after one period, all data will delete to avoid data overload

Reply:
understand how refreshInterval work it fine, but please careful when 5000 ml the api will call and upadte render list will slow for performance I the page alway re-render after get api

Reply:
Yes, I changed it to make sure API does't call continuously after 30 minutes
![image](https://github.com/datdragon777/React-Training/assets/74806826/2e4225b2-9e15-4d79-aedb-1477520188b1)